src/indieweb_utils/replies/context.py

```diff
@@ -29,298 +40,316 @@ def get_reply_context(url, twitter_bearer_token=None):
     :rtype: list
     """
 
-    h_entry = None
-    photo_url = None
-    site_supports_webmention = False
+    return_object = ReplyContext
+    photo_url = ""
     parsed_url = url_parse.urlsplit(url)
     http_headers = {"Accept": "text/html", "User-Agent": "indieweb_utils"}
 
-    if url.startswith("https://") or url.startswith("http://"):
-        try:
-            page_content = requests.get(url, timeout=10, verify=False, headers=http_headers)
-        except:
-            return True, {"error": "Page content could not be retrieved."}
+    if not url.startswith("https://") and not url.startswith("http://"):
+        return return_object
 
-        if page_content.status_code != 200:
-            return True, {"error": page_content.status_code}
+    try:
+        page_content = requests.get(url, timeout=10, verify=False, headers=http_headers)
+    except:
+        raise ReplyContextRetrievalError("Could not retrieve page content.")
 
-        site_supports_webmention, _ = discover_webmention_endpoint(url)
+    if page_content.status_code != 200:
+        raise ReplyContextRetrievalError("Page did not return a 200 response.")
```

Comment:
It would be helpful to put the actual returned status code in the error message.

🤔  I wonder if we there aren't any special cases we'd want to handle (as a user of the library) a particular status code _different_. I'm thinking mostly like a 500 or 404?  Just a thought, no need to make changes.